src/ImageSharp/Processing/AffineTransformBuilder.cs

```diff
@@ -310,6 +323,32 @@ public Matrix3x2 BuildMatrix(Rectangle sourceRectangle)
         return matrix;
     }
 
+    /// <summary>
+    /// Returns the size of a rectangle large enough to contain the transformed source rectangle.
+    /// </summary>
+    /// <param name="sourceRectangle">The rectangle in the source image.</param>
+    /// <exception cref="DegenerateTransformException">
+    /// The resultant matrix is degenerate containing one or more values equivalent
+    /// to <see cref="float.NaN"/> or a zero determinant and therefore cannot be used
+    /// for linear transforms.
+    /// </exception>
+    /// <returns>The <see cref="Size"/>.</returns>
+    public Size GetTransformedSize(Rectangle sourceRectangle)
+    {
+        Size size = sourceRectangle.Size;
+
+        // Translate the origin matrix to cater for source rectangle offsets.
+        Matrix3x2 matrix = Matrix3x2.CreateTranslation(-sourceRectangle.Location);
+
+        foreach (Func<Size, Matrix3x2> factory in this.boundsMatrixFactories)
```

Comment:
```suggestion
        Matrix3x2 matrix = Matrix3x2.CreateTranslation(-sourceRectangle.Location);
        matrix.Translation += new Vector2(1);

        foreach (Func<Size, Matrix3x2> factory in this.transformMatrixFactories)
        {
```

I believe this change should prevent the need of having the 2nd collection of `Func`s as I think matrix's for sizing should originate at (+1,+1) from the location of the transform matrix.

Reply:
You’ve got me second guessing myself now but I’m not sure that would work. For the most part we want to return the transformed extents based on the explicitly provided information (translate/rotate/scale x, y). It’s only when we perform our own maths to calculate the centre point we go astray. That matrix must be zero based but can then not be used for bounds calculation.

Reply:
Confirmed. This example would add an additional unwanted pixel to the right side of the image with the suggested changes.

![image](https://github.com/SixLabors/ImageSharp/assets/385879/2fcb9dc6-3d75-4ad7-9062-3179ffeeebe6)

Reply:
thanks for validating, just want to make sure we hadn't missed something more simple, this PR looks fine to me then. 👍